ENH Allow `SelectFromModel`'s `max_features` to accept callables (#22356)
scikit-learn/scikit-learn: doc/whats_new/v1.1.rst

- |Efficiency| Improve runtime performance of :func:`feature_selection.chi2`
  with boolean arrays. :pr:`22235` by `Thomas Fan`_.

- |Feature| Added the ability to pass floats and callables to the `max_features`
  parameter of :class:`feature_selection.SelectFromModel`. Also introduced new
  attribute `max_features_` which is inferred from `max_features` and the data

>   during `fit`. If `max_features` is integral, then `max_features_=max_features`.

Reviewer: ```suggestion
  during `fit`. If `max_features` is a int, then `max_features_ = max_features`.
```

>   If `max_features` is a float, then `max_features_=round(max_features*X.shape[1])`.

Reviewer: ```suggestion
```

>   If `max_features` is a callable, then `max_features_=max_features(X)`.

Reviewer: ```suggestion
  If `max_features` is a callable, then `max_features_ = max_features(X)`.
```
